text/0000-explode.md

```diff
@@ -0,0 +1,242 @@
+
+Therefore, we propose new metadata in the `ember-addon` section of `package.json`:
+
+```js
+  "ember-addon": {
+    "required-core-features": "0.4.0",
+    "required-additional-features": ["@ember/routing", "@ember/prototype-etxensions", "@ember/string"]
+  }
+```
+
+`required-core-features` records the latest version of `@ember/core` that the addon has been tested against. It causes your addon to depend on all the Ember packages that were still bundled inside that `@ember/core` version. 
+
+`required-additional-features` lets the addon express dependency on features that have already been split from `@ember/core` as of the base features version.
+
+
+### Why required-core-features instead of peerDependencies?
+
+`required-core-features` says something different than a `peerDependency` constraint.
+
+We want addons to be as aggressive as possible in bumping their `required-core-features`, because that creates the maximum opportunity for apps to pick their packages *a la carte*.
+
+Simultaneously, we want addons to support the widest practical range of Ember versions (which would be their `peerDependencies` constraint on Ember, if they choose to have one), because that makes everybody's upgrades easier. 
+
+For example, consider an addon with:
+
+```js
+  "ember-addon": {
+    "required-core-features": "0.4.0",
+    "required-additional-features": ["@ember/routing"]
+  }
+```
+
+If you use this addon in an app with `ember-source` 2.18.0, there's no problems, the metadata isn't affecting anything.
+
+If you use this addon in an app with `@ember/core` 0.6.0, there's still no problems, but we will be able to warn the user that they need to install `@ember/routing` plus whatever future packages happen to get split out of `@ember/core` between 0.4.0 and 0.6.0.
+
+In contrast, if there was a peerDependency on `@ember/core` we would break usage with `ember-source`, and *vice versa*. And if there was a peerDependency on `@ember/routing` it would break usage with any Ember version earlier than the one that split out that package.
+
+The "pure NPM" way of solving this problem is to release new versions of your addon to support new versions of Ember, but this creates a much more fragmented ecosystem than we want. We are careful not to break things very often, such that many addons really don't need to be released just to keep working with newer Ember releases.
+
+
+## Local and Crowd-sourced Dependency Overrides
+
+We have already proposed making missing dependencies a hard build error. This is important so that weird and frustrating bugs don't crop up unexpectedly. But it needs to be possible to get oneself unstuck without being blocked by third-party code.
+
+Therefore, we propose a `config/dependency-overrides.js` file that allows you to make declarations on behalf of any addon:
```

Comment:
This was also a topic of discussion in #649. @wycats was of the opinion that overriding an addon with config is not the right path to promote. If an addon is out of date, the incentives should be to go fix the addon and use a soft fork in the meantime.